Cargo.toml

```diff
@@ -16,16 +16,19 @@ edition = "2018"
 bitvec = "0.22"
 blake2s_simd = "0.5"
 ff = "0.10"
-futures = "0.1"
-futures-cpupool = { version = "0.1", optional = true }
 group = "0.10"
-num_cpus = { version = "1", optional = true }
-crossbeam = { version = "0.7", optional = true }
 pairing = { version = "0.20", optional = true }
 rand_core = "0.6"
 byteorder = "1"
 subtle = "2.2.1"
 
+# Multicore dependencies
+crossbeam-channel = { version = "0.5", optional = true }
+lazy_static = { version = "1.4.0", optional = true }
+log = { version = "0.4.8", optional = true }
+num_cpus = { version = "1", optional = true }
+rayon = { version = "1.3.0", optional = true }
```

Comment:
```suggestion
crossbeam-channel = { version = "0.5.1", optional = true }
lazy_static = { version = "1.4.0", optional = true }
log = { version = "0.4.14", optional = true }
num_cpus = { version = "1.13", optional = true }
rayon = { version = "1.5.1", optional = true }
```

Reply:
Went do we need to

Reply:
Why do we need to set these specific minimum versions, instead of allowing the downstream package managers to resolve them? Is there something specific we rely upon from these versions?

Reply:
This is the version of rayon used by https://github.com/zcash/halo2/blob/main/Cargo.toml . We don't have to specify minor version numbers, but that's what we've been doing in other crates, and we should be consistent (in order to minimise the use of multiple versions of crates in the same executable, which produces larger executables and is potentially confusing for debugging).

In the particular case of rayon, using multiple versions will result in multiple global thread pools, I think, which would be less efficient if you were doing a bellman proof and a halo2 proof at the same time.

Reply:
Also apparently `crossbeam-deque 0.8.0`, which is depended on by `rayon 0.5.0`, had a security vulnerability: https://github.com/rayon-rs/rayon/pull/876#issue-704335337 . zcashd currently pins non-vulnerable version 0.7.3 of the crossbeam crates (else I would not mention this in a public comment), but we want the next update to skip to 0.8.1. I don't know what the security bug was.

Reply:
I think the bug was probably https://github.com/crossbeam-rs/crossbeam/pull/726

Reply:
Note that `crossbeam-channel 0.5.1` [depends on `num_cpus 1.13.0`](https://github.com/crossbeam-rs/crossbeam/blob/master/crossbeam-channel/Cargo.toml). We could relax the dependency to `num_cpus 1` I guess. Note that `num_cpus` does not appear to be following semver, because [it's updating the minor version for compatible bug fixes](https://github.com/seanmonstar/num_cpus/blob/master/CHANGELOG.md).)

For `log` we could relax the dependency to `0.4` (not `0`, since [there were many API changes relative to 0.3.x](https://github.com/seanmonstar/num_cpus/blob/master/CHANGELOG.md). `log` does appear to follow semver.